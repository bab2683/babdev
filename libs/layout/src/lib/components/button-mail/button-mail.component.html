<a href="mailto:{{ value }}">
  <fa-icon [icon]="icon"></fa-icon>
</a>
