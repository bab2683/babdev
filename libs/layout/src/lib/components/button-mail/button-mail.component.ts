import { Component } from '@angular/core';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { faEnvelope } from '@fortawesome/free-regular-svg-icons';

@Component({
  selector: 'babdev-button-mail',
  templateUrl: './button-mail.component.html',
  styleUrls: ['./button-mail.component.scss']
})
export class ButtonMailComponent {
  public icon: IconDefinition = faEnvelope;
  public value: string;
}
