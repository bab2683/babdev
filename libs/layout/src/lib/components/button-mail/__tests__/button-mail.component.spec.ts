import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ButtonMailComponent } from '../button-mail.component';

describe('ButtonMailComponent', () => {
  let component: ButtonMailComponent;
  let fixture: ComponentFixture<ButtonMailComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ButtonMailComponent],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ButtonMailComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
