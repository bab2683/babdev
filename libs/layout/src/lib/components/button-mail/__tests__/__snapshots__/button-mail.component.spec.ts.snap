// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonMailComponent should match snapshot 1`] = `
<babdev-button-mail
  icon={[Function Object]}
>
  <a
    href="mailto:"
  >
    <fa-icon />
  </a>
</babdev-button-mail>
`;
